util/system/path.go

```diff
@@ -16,3 +25,180 @@ func DefaultPathEnv(os string) string {
 	}
 	return DefaultPathEnvUnix
 }
+
+// NormalizePath functions similarly to filepath.Join(), but with a few distinctions:
```

Comment:
Maybe `NormalizedJoin()`? Otherwise, I would not expect it take 2 input paths.

For the platform, the name should signify that it is the input platform.

Reply:
This function is a variation of: https://github.com/moby/buildkit/blob/master/client/llb/fileop.go#L772-L789 with added OS awareness. It was augmented in order to be useful in more cases. There are a number of places throughout the code where the same pattern repeats. 

The PR that makes use of the functions here, has a number of examples: https://github.com/moby/buildkit/pull/3908/files

The fact that this function does more than join is the reason I borrowed the name of the function that inspired it.

Reply:
Ok, remove the comparisons with `Join` and explain that "if path is relative, it is appended to the parent path."